mitmproxy/certs.py

```diff
@@ -337,10 +363,54 @@ def dummy_cert(
     # https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.2 states
     # that SKI is optional for the leaf cert, so we skip that.
 
+    fake_distribution_points = []
+    for upstream_distribution_point in revocation.crl_distribution_points:
+        upstream_url = urllib.parse.urlparse(upstream_distribution_point)
+        # Replace original URL path with the CA cert serial number, which acts as a magic token
+        fake_url = upstream_url._replace(
```

Comment:
I think there's a subtle difference between this and `request.path.endswith()` in tlsconfig: `request.path` may include a query string. https://github.com/mitmproxy/mitmproxy/blob/main/mitmproxy/net/http/url.py#L20 is the better alternative here.

Reply:
Switched to the parse function in url.py as recommended